### PR TITLE
Misc. cleanup and fixes while working on discretization

### PR DESCRIPTION
**Pull request by pkjha-aero**

### CODE REVIEW

File: Exec/ScalarAdvection/prob.H [hunk 1]
```diff
@@ -49,7 +49,7 @@ pc_initdata(
   // Set the density 
   state(i, j, k, Density_comp) = ProbParm::rho_0;
  
-  // Initial temperature -- though we should make sure its consistent with (rho e) above
+  // Initial potential temperature -- though we should make sure it's consistent with (rho e) above
```
Comment (ewquon) [inline Exec/ScalarAdvection/prob.H]: @pkjha-aero , what `rho e` are you referring to?

Comment (pkjha-aero) [reply Exec/ScalarAdvection/prob.H]: @ewquon @jmirocha. The original comment was by Ann. I just added 'potential'. But, now that we have our equation for (rho theta), this comment about rho*e or rho*E is no longer relevant. So, we can get rid of this.

Comment (pkjha-aero) [reply Exec/ScalarAdvection/prob.H]: These comments/edits were from days before we had our governing equations and discretizations worked out and documented.

**Review by ewquon — APPROVED** [review]

**Comment by ewquon** [comment]
@brigaudet this latest merged PR includes the ref pressure fix, FYI

**Comment by brigaudet** [comment]
Thanks! - looks good.

--Brian

**Pull request merged**